aptos-move/block-executor/src/errors.rs

```diff
@@ -2,62 +2,47 @@
 // Parts of the project are originally copyright © Meta Platforms, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-use aptos_aggregator::types::PanicOr;
-use aptos_logger::{debug, error};
-use aptos_mvhashmap::types::TxnIndex;
 use aptos_types::delayed_fields::PanicError;
-use aptos_vm_logging::{alert, prelude::*};
+
+// The same module access path for module was both read & written during speculative executions.
+// This may trigger a race due to the Move-VM loader cache implementation, and mitigation requires
+// aborting the parallel execution pipeline and falling back to the sequential execution.
+// TODO: provide proper multi-versioning for code (like data) for the cache.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ModulePathReadWriteError;
+
+// This is not PanicError because we need to match the error variant to provide a specialized
+// fallback logic if a resource group serialization error occurs.
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ResourceGroupSerializationError;
```

Comment:
yeah, this is purely for the serialize_groups function, because it is called both in sequential and parallel.

I wanted to make every function return only errors it can create.